### DISCUSSION TITLE
Synchronizing Optitrack positioning data with Crazyflie logging

### DISCUSSION THREAD

**Discussion by suvansh** [bitcraze/discussions#206] [Q&A] [answered]
When exclusively using crazyflie logging (with the crazyflie loco positioning setup), I can log the timestamp and parameter groups (across separate LogConfigs) to a CSV, allowing me to sync up values across parameter groups with the timestamps. I'm considering switching to optitrack for position/attitude, but would still be using crazyflie's linear/angular velocity logging. My concern is how to synchronize the data from these two streams (Optitrack and Crazyflie logging).
One solution that comes to mind is to continue to use crazyflie's position/attitude logging and use that as a means of aligning the data with the optitrack data, but I'm wondering if there's a more elegant solution to this problem.

**Comment by krichardsson** [comment, marked as answer]
HI!

> One solution that comes to mind is to continue to use crazyflie's position/attitude logging and use that as a means of aligning the data with the optitrack data

This is probably the easiest solution and would give you pretty good time stamping.

The solution depends on what kind of precision you need. The logging framework is a bit of "best effort", it will transmit data (and sample variables) when there is space in an outgoing packet. In most cases this is pretty instantaneous, but it is not guaranteed. The time stamp you get on the python side is when the packet arrives (I think). This means that the time stamp probably is pretty accurate, but the sample rate is possibly not very precise or stable. 

If you need better time stamping, you could use the SD-card deck and log to an SD-card on the Crazyflie. That will provide better time stamping, but it is less convenient as you have to move the SD-card back and forth.

**Reply by suvansh** [reply]
Great, thanks so much for the explanation and suggestions! Will give that a shot with some sort of correlation between the crazyflie and Optitrack data.
